.github/workflows/iroha2-pr.yml

```diff
@@ -8,46 +8,131 @@ env:
+        if: always()
+      - name: Cleanup test environment
+        run: bash -c './scripts/test_env.sh cleanup docker'
+
+
+  bench:
+    runs-on: [self-hosted, Linux]
+    container:
+      image: 7272721/i2-ci:latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: Swatinem/rust-cache@v1
+      - name: Run benchmarks
+        run: mold -run cargo bench --workspace --quiet
+
+# ------------------------------ SDK tests go here ------------------------
+
+  java-api-test:
+    runs-on: [self-hosted, Linux] #ubuntu-latest
+    container:
+      image: 7272721/i2-ci:latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.11
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
         with:
-          toolchain: 1.56
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Build Iroha Peer
+        run: mold -run cargo build
+        working-directory: cli
       - name: Build Client CLI
-        run: cargo build --verbose
+        run: mold -run cargo build
         working-directory: client_cli
-      - name: Setup docker test environment
-        run: ./scripts/setup_docker_test_env.sh
-      - name: Docker compose genesis test
-        run: ./scripts/test_genesis_docker_compose.sh
-      - name: Docker compose test
-        run: ./scripts/test_docker_compose.sh
-      - name: Cleanup docker test environment
-        run: ./scripts/cleanup_docker_test_env.sh
+      - name: Mark both binaries as executable
+        run: |
+          chmod +x target/debug/iroha
+          chmod +x target/debug/iroha_client_cli
+      - name: Setup test environment
+        run: ./scripts/test_env.sh setup
+      - name: Test iroha2-java API
+        run: ./scripts/API/java.sh setup && ./scripts/API/java.sh run
+      - name: Cleanup test iroha2-java API
+        run: ./scripts/API/java.sh cleanup
+      - name: Cleanup test environment
+        run: ./scripts/test_env.sh cleanup
+      - name: Cleanup Gradle Cache
+        # Remove some files from the Gradle cache, so they aren't
+        # cached by GitHub Actions.  Restoring these files from a
+        # GitHub Actions cache might cause problems for future builds.
+        run: |
+          rm -f ~/.gradle/caches/modules-2/modules-2.lock
+          rm -f ~/.gradle/caches/modules-2/gc.properties
+
+  long-tests:
+    runs-on: [self-hosted, Linux] #ubuntu-latest
+    container:
+      image: 7272721/i2-ci:latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run long tests
+        run: cargo test --workspace --no-fail-fast -- --ignored --test-threads=1 long
```

Comment:
these tests are prefixed `long`?

Reply:
Not exactly, but it's good enough, until we introduce a better testing fixture.